src/logic/calculate.js

```diff
@@ -2,16 +2,31 @@ import operate from './operate';
 
 const calculate = ({ total, next, operation }, bName) => {
   if (total !== null && next !== null && operation !== null) {
-    if (Number.isNaN(bName)) {
+    const isNotNumber = str => !str.match(/\d/);
+    if (total === 'Infinity') {
+      return {
+        total: '0',
+        next: '',
+        operation: '',
+      };
+    }
+    if (isNotNumber(bName)) {
       if (bName === '+/-') {
         return {
           total: parseFloat(total) ? (-1 * total).toString() : total,
           next: parseFloat(next) ? (-1 * next).toString() : next,
           operation: '',
         };
       } if (bName === '+' || bName === '-' || bName === 'X' || bName === '÷') {
+        if (next !== '' && total !== '' && operation !== '') {
+          return {
+            total: operate(total, next, operation),
+            next: '',
+            operation: bName,
+          };
+        }
         return {
-          total: next,
+          total: (next === '') ? total : next,
           next: '',
           operation: bName,
         };
```

Comment:
Whenever the `AC` button is  clicked, we should have `0` as the result right.... shown in the loom video